cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java

```diff
@@ -337,7 +336,7 @@ private void handlePartitionDeletion(PartitionUpdate pu, OffsetPosition offsetPo
         if (!clusteringColumns.isEmpty()) {
             throw new CassandraConnectorSchemaException("Uh-oh... clustering key should not exist for partition deletion");
         }
-        Collection<ColumnMetadata> columns = tableMetadata.getColumns().values();
+        List<ColumnMetadata> columns = new ArrayList<>(tableMetadata.getColumns().values());
```

Comment:
@jpechane @vjuranek this is the bug I was talking about in JIRA.

Reply:
@vjuranek without this fix in the tests would fail so do I get it right we need to fix this first before this one can go in?

Reply:
Yes. No problem to send it in one PR, but it would be nice to split it into separate commits (one fixing this issue, second one adding `onKeyspaceCreated()` and the last one for refactoring).

Reply:
@vjuranek done, it is building.

Reply:
I think I still need to create these separate jira tickets though :)) I ll do it.

Reply:
@vjuranek it is now fully done, DBZ-5022 and DBZ-5023.